Libraries/LibGUI/AboutDialog.cpp

```diff
@@ -97,6 +98,13 @@ AboutDialog::AboutDialog(const StringView& name, const Gfx::Bitmap* icon, Window
         make_label("SerenityOS");
     make_label(version_string());
     make_label("Copyright \xC2\xA9 the SerenityOS developers, 2018-2020");
+    make_label("Licensed under the BSD 2-Clause License.");
+
+    auto& link_label = right_container.add<LinkLabel>("http://serenityos.org");
```

Comment:
Here's a little problem: this dialog is not only shown in the About app, it's shown in virtually every app.

Reply:
Hi,
Uppss this is a big problem :( @awesomekling what do you prefer for this.
I prefer drop commits for about dialog but i cant find other use case for this component for now.

Reply:
Yeah this is a big problem that will eventually require some kind of architectural decision to be made. @bugaevc what do you think we should do here?

Reply:
(I assume you're talking about the same thing I was — that now basically every app requires access to the launch server.)

That's actually a deeper problem with `pledge()` and `unveil()`: they affect the process as a whole, but you don't control the whole process if you're using some library (and you always do), and that library expects to make use of some standard functionality. The file dialog is another example: every app that wants to display the file dialog now has to pledge thread simply because the dialog uses threading. Also everyone needs Unix and accept and whatnot just because of the RPC server in LibCore.

I honestly don't know how we should solve the problem on the fundamental level. The file picker should be moved out of process; and for the link, perhaps we can only show it in the system about box and not in every app.

Reply:
Can we create WindowServer API for common dialog boxes?
In this approach, all permissions go to WindowServer and we don't need to add any lib to the application.

Reply:
WindowServer cannot use LibGUI, so it wouldn't be a good place to implement these. But perhaps a DialogServer service that provides AboutDialog, FilePicker, etc..

Reply:
I am trying to implement the DialogServer approach. With this solution, we should add 'pledge("unix")' and 'unveil("/tmp/portal/dialog")' to the application that wants to use a DialogServer.

At this point, we are at the same point as we are trying to avoid.